hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClientDynamicClusterConfig.java

```diff
@@ -134,6 +135,9 @@ public Config addMapConfig(MapConfig mapConfig) {
         Data partitioningStrategy = mapConfig.getPartitioningStrategyConfig() == null
                 ? null : serializationService.toData(mapConfig.getPartitioningStrategyConfig().getPartitioningStrategy());
 
+        DataPersistenceAndHotRestartMerger
```

Comment:
Merging hot-restart/data-persistence configs should also happen here, like already done in superclass methods `Config#addCacheConfig` and `Config#addMapConfig`. This is also a workaround for https://github.com/hazelcast/hazelcast-enterprise/issues/4282

Reply:
ohh, nicee, thank you :)